src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -147,15 +148,17 @@ void Terminal::UpdateSettings(ICoreSettings settings)
     // size is smaller than where the mutable viewport currently is, we'll want
     // to make sure to rotate the buffer contents upwards, so the mutable viewport
     // remains at the bottom of the buffer.
-    if (_buffer)
+
+    // Regenerate the pattern tree for the new buffer size
+    if (_mainBuffer)
     {
         // Clear the patterns first
-        _buffer->ClearPatternRecognizers();
+        _mainBuffer->ClearPatternRecognizers();
         if (settings.DetectURLs())
         {
             // Add regex pattern recognizers to the buffer
             // For now, we only add the URI regex pattern
-            _hyperlinkPatternId = _buffer->AddPatternRecognizer(linkPattern);
+            _hyperlinkPatternId = _mainBuffer->AddPatternRecognizer(linkPattern);

```

Comment:
now here's a fun trick! URLs inside the alt buffer will no longer be parsed.

Reply:
we don't populate the matching settings when we create it, and we don't do it when settings are updated while it is active... hmmmm

Reply:
fixed